### DISCUSSION TITLE
구글캘린더 기능 아이디어 입니다!

### DISCUSSION THREAD

**Discussion by thiat1998** [DevAnyKR/StreamDock.Plugin.GoogleAPI#14] [Ideas]
잘 사용중에 있으나, 역시 표출되어지는 칸이 작아선지 아쉬움이 있군요..
그래서 생각해본 아이디어가
1. 폰트크기 수정기능
2. 일정이 두개 이상일시 각각의 일정을 고를수 있게 하는 기능

이렇게 두 기능이 있으면 정말 좋을것 같습니다!

덕분에 한글패치와 구글 캘린더 기능 추가하였습니다 감사드립니다!

![KakaoTalk_20240620_143233160](https://github.com/DevAnyKR/StreamDock.Plugin.GoogleAPI/assets/112832077/59cf1001-d143-4e8f-9925-80ae94705157)

**Comment by DevAnyKR** [comment]
> 폰트크기 수정기능

처음에 글꼴을 줄여서 표시해봤는데, 글자가 너무 작으면 읽기가 어렵고 크면 잘리고, 그래서 크기 조절 기능을 추가 예정이었습니다....만 자잘한 버그가 너무 많아서 새 기능을 추가하는데 시간이 걸려요 😢
현재 적용된 폰트는 Arial, 32px, Bold 입니다. 
출력 목록은 현재 5개로 설정해놨습니다.

> 일정이 두개 이상일시 각각의 일정을 고를수 있게 하는 기능

계정은 로그인을 하면 되지만, 달력 그룹은 PI 설정 화면으로 목록을 피드백 줘야합니다. 이부분은 아직 테스트 중이에요. 언제될지...🤣
달력에 설정된 색상으로 구분할 계획입니다.
현재는 primary 일정만 보여줍니다.


AJAZZ 제품도 스트림독 프로그램으로 293S인식해서 동작하나요?
몇달전 주문할땐 중국용만 있다해서 판매거부 당했었어요;지금은 이름없는 묻지마 제품 씁니다.

**Reply by thiat1998** [reply]
오 출력목록이 늘어났군요! 감사합니다!!

AJAZZ 제품은 스트림독 프로그램으로 인식이 안되더군요...
프로그램 구성은 ajazz꺼나 스트림독이나 아예 동일합니다~!

스트림독 프로그램보단 업데이트는 늦는것 같네요 ㅠㅠ

헉 그러시군요.. 판매거부까지ㄷㄷ...

![스크린샷 2024-06-21 130901](https://github.com/DevAnyKR/StreamDock.Plugin.GoogleAPI/assets/112832077/edfd068b-cbde-436b-b088-87af0694ad53)

**Reply by DevAnyKR** [reply]
스트림독은 며칠전에 v2.10.179.426으로 업데이트 되었습니다. 버그가 좀 있지만...
아마도 제조·개발사에서 만들고 각 판매사에 OEM으로 배포하나봅니다.

스트림독에는 위에 포럼 아이콘과 유투브 튜토리얼 아이콘이 있습니다.
![image](https://github.com/DevAnyKR/StreamDock.Plugin.GoogleAPI/assets/110871727/692de4a8-f892-49e4-9cc7-7648b13e9bce)

**Reply by thiat1998** [reply]
저는 2.9.177.122가 마지막 업데이트군요..
맞습니다 그런듯합니다 국내 몬스타덱도 동일한 느낌이더라구요

신기하네요 왼쪽아래 기기이미지도 나오는게 신기합니다 ㅋㅋ

**Reply by DevAnyKR** [reply]
서일대...서울에 있는 그곳이라면 제가 다녔던 곳이에요 😮
처음에 글자 길이를 확인해서 너비에 맞게 줄이도록 했는데, 긴 제목은 너무 줄어들어서 안 보이더라고요.
그래서 크기를 고정했습니다.

1초틱 이벤트나 100ms 이상의 주기로 이미지를 계속 갱신해서 스트림독에 출력하는 기능을 완성하면
**일정 좌우 스크롤 무빙** 효과가 가능...할거라 믿습니다.
훨씬 부드러운 10ms주기 갱신은 윈도우가 버벅거리더라고요.

**Reply by thiat1998** [reply]
헉..?? 저도 서일대 출신인데 신기하네요 ㅋㅌㅋ 여기서 동문을 다 만난다는게 정말 신기합니다!!

오우 좌우 스크롤 무빙 정말 좋은 아이디어인듯 하네요 그러면 다 보이기도 하겠네요

**Reply by DevAnyKR** [reply]
좌 스크롤 기능 넣어봤습니다. 0.5초 단위로 이동합니다. 부드럽게하면 다른 키가 먹통이 되는데 방법을 몰라요🤣

**Reply by thiat1998** [reply]
오 기대가 되는군요 내일 회사가서 빨리 싸봐야겠습니다~!

> 좌 스크롤 기능 넣어봤습니다. 0.5초 단위로 이동합니다. 부드럽게하면 다른 키가 먹통이 되는데 방법을 몰라요🤣
